src/pyTRLCConverter/rst_converter.py

```diff
@@ -0,0 +1,701 @@
+        """
+        result = Ret.OK
+
+        # Multiple document mode?
+        if self._args.single_document is False:
+            assert self._fd is None
+
+            file_name_rst = self._file_name_trlc_to_rst(file_name)
+            result = self._generate_out_file(file_name_rst)
+
+            # The very first written reStructuredText part shall not have an empty line before.
+            self._empty_line_required = False
+
+        return result
+
+    def leave_file(self, file_name: str) -> Ret:
+        # lobster-trace: SwRequirements.sw_req_rst_multiple_doc_mode
+        """
+        Leave a file.
+
+        Args:
+            file_name (str): File name
+
+        Returns:
+            Ret: Status
+        """
+
+        # Multiple document mode?
+        if self._args.single_document is False:
+            assert self._fd is not None
+            self._fd.close()
+            self._fd = None
+
+        return Ret.OK
+
+    def convert_section(self, section: str, level: int) -> Ret:
+        # lobster-trace: SwRequirements.sw_req_rst_section
+        """
+        Process the given section item.
+        It will create a reStructuredText heading with the given section name and level.
+
+        Args:
+            section (str): The section name
+            level (int): The section indentation level
+        
+        Returns:
+            Ret: Status
+        """
+        assert len(section) > 0
+        assert self._fd is not None
+
+        self._write_empty_line_on_demand()
+        rst_heading = self.rst_create_heading(section,
+                                            self._get_rst_heading_level(level),
+                                            os.path.basename(self._fd.name))
+        self._fd.write(rst_heading)
+
+        return Ret.OK
+
+    def convert_record_object_generic(self, record: Record_Object, level: int) -> Ret:
+        # lobster-trace: SwRequirements.sw_req_rst_record
+        """
+        Process the given record object in a generic way.
+
+        The handler is called by the base converter if no specific handler is
+        defined for the record type.
+
+        Args:
+            record (Record_Object): The record object
+            level (int): The record level
+        
+        Returns:
+            Ret: Status
+        """
+        assert self._fd is not None
+
+        self._write_empty_line_on_demand()
+
+        return self._convert_record_object(record, level, None)
+
+    def finish(self):
+        # lobster-trace: SwRequirements.sw_req_rst_single_doc_mode
+        """
+        Finish the conversion process.
+        """
+
+        # Single document mode?
+        if self._args.single_document is True:
+            assert self._fd is not None
+            self._fd.close()
+            self._fd = None
+
+        return Ret.OK
+
+    def _write_empty_line_on_demand(self) -> None:
+        # lobster-trace: SwRequirements.sw_req_rst
+        """
+        Write an empty line if necessary.
```

Comment:
What necessary is should be explained here. 
I think there was a good comment at a call site of this function, but it should be in the doc string here.

Reply:
ok